.github/workflows/build.yml

```diff
@@ -6,21 +6,205 @@ on:
+    env:
+      MAKEFLAGS: "-j 2"
+      CONFIGURE_FLAGS: ${{ matrix.config_flags }}
     steps:
     - uses: actions/checkout@v2
+    - run: type pkg-config
+    - run: pkg-config --list-all | sort -u
     - name: Generate configure script
       run:
         ./build.sh
     - name: configure
-      run: ./configure
+      run: ./configure $CONFIGURE_FLAGS
+    - name: Make
+      run: make
+    - name: make check
+      run: make check
+    - name: install bzip2
+      continue-on-error: true
+      run: |
+        yum install -y bzip2 || apt install -y bzip2
+    - name: make
+      continue-on-error: true
+      run: |
+        make distcheck DISTCHECK_CONFIGURE_FLAGS="--disable-dependency-tracking --enable-debug"
+
+  experimental-fedora:
+    runs-on: ubuntu-20.04
+    container: ${{ matrix.container_image }}
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # for tasks that are optional, use the continue-on-error option, to prevent a workflow from failing when the task fails
+        container_image:
+          - fedora:34
+          - fedora:rawhide
+        experimental: [ true ]
+        install_packages: [ true ]
+        config_flags: [ "--disable-dpdkstat --disable-dpdkevents --disable-virt --disable-xmms"]
+    env:
+      MAKEFLAGS: "-j 2"
+      CFLAGS: "-fPIE -Wno-deprecated-declarations"
+      CPPFLAGS: "-fPIE -Wno-deprecated-declarations"
+      CONFIGURE_FLAGS: ${{ matrix.config_flags }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install build requirements
+      run: |
+        dnf install -y \
+        autoconf \
+        automake \
+        bison \
+        clang \
+        cpp \
+        curl \
+        flex \
+        gcc \
+        gcc-c++ \
+        gdb \
+        git \
+        glibc-devel \
+        libgcrypt-devel \
+        libtool \
+        libtool-ltdl-devel \
+        m4 \
+        make \
+        nc \
+        pkgconfig \
+        which
+    - name: Install dependencies
+      run: |
+        dnf install -y \
+        OpenIPMI-devel \
+        dpdk-devel \
+        ganglia-devel \
+        gpsd-devel \
+        gtk2-devel \
+        hiredis-devel \
+        intel-cmt-cat-devel \
+        iproute-devel \
+        iptables-devel \
+        java-11-openjdk-devel \
+        java-devel \
+        jpackage-utils \
+        libatasmart-devel \
+        libcap-devel \
+        libcurl-devel \
+        libdbi-devel \
+        libesmtp-devel \
+        libmemcached-devel \
+        libmicrohttpd-devel \
+        libmnl-devel \
+        libmodbus-devel \
+        libnotify-devel \
+        liboping-devel \
+        libpcap-devel \
+        librabbitmq-devel \
+        librdkafka-devel \
+        libsigrok-devel \
+        libudev-devel \
+        libvirt-devel \
+        libxml2-devel \
+        lm_sensors-devel \
+        lua-devel \
+        lvm2-devel \
+        mosquitto-devel \
+        mysql-devel \
```

Comment:
mariadb-devel, mysql-devel does not exist in Fedora 34+

Reply:
Not sure if that actually causes the node error, but it has to be fixed anyways.

Reply:
It is undetermined what is causing the error on fedora:rawhide, since the packages were not installed, due to being unable to resolve some link.

On both Fedora 34 and rawhide, ``dnf install mysql-devel`` will install ``mariadb-devel``, so this is most likely not causing the error. However, I have updated this dependency, because it is misleading anyone who casually looks at this.